modules/aioseop_sitemap.php

```diff
@@ -403,7 +446,175 @@ public function __construct() {
+					if ( empty( $all_terms ) ) {
+						wp_send_json_error( array( 'msg' => __( 'No terms found', 'all-in-one-seo-pack' ) ) );
+					}
+					$response = $all_terms;
+					break;
+			}
+			wp_send_json_success( $response );
+		}
+
+		/**
+		 * Generates the values to be displayed as pre-selected in the exclude terms box.
+		 *
+		 * @param array $terms Array of term_ids.
+		 *
+		 * @since 3.0
+		 *
+		 * @return array Array of term_id vs term_name/term_taxonomy combination.
+		 */
+		private function get_terms_as_selected( $terms ) {
+			$all_terms = array();
+			foreach ( $terms as $id ) {
+				$term = get_term( (int) $id );
+				if ( ! $term ) {
+					continue;
+				}
+				$all_terms[ $id ] = sprintf( '%s (%s)', $term->name, $term->taxonomy );
+			}
+			return $all_terms;
+		}
+
+
+		/**
+		 * Checks if a taxonomy has been excluded. This could only be because all its terms have been selected to be excluded.
+		 *
+		 * @since 3.0
+		 *
+		 * @param string    $tax    The taxonomy name.
+		 *
+		 * @return bool Whether a taxonomy has a been excluded.
+		 */
+		private function has_taxonomy_been_excluded( $tax ) {
+			global $wp_version;
+			$term_count = intval( wp_count_terms( $tax, array( 'hide_empty' => true ) ) );
+			if ( 0 === $term_count ) {
+				// no terms in this taxonomy, so not excluded.
+				return false;
+			}
+
+			// terms excluded in the settings.
+			$terms_excluded = $this->options[ "{$this->prefix}excl_terms" ];
+			if ( empty( $terms_excluded ) ) {
+				// no terms excluded from any category, so not excluded.
+				return false;
+			}
+
+			// terms defined for this taxonomy.
+			$terms  = get_terms(
+				array(
+					'taxonomy' => $tax,
+					'hide_empty' => true,
+				)
+			);
+			if ( is_wp_error( $terms ) ) {
+				return false;
+			}
+			$term_ids = array();
+			foreach ( $terms as $term ) {
+				$term_ids[] = $term->term_id;
+			}
+			// are all terms from this taxonomy excluded?
+			$common = count( array_intersect( $term_ids, $terms_excluded ) );
+			// the category is excluded if ALL of its terms have been specified in the exclude terms list.
+			return $common === count( $term_ids );
+		}
+
+		/**
+		 * Load resources.
+		 *
+		 * @since 3.0
+		 *
+		 * @param string $hook_suffix The current admin page.
+		 */
+		function admin_enqueue_scripts( $hook_suffix ) {
+			$deps = array();
+			$options = array(
+				'ajax' => array(
+					'action' => $this->prefix,
+					'nonce' => wp_create_nonce( $this->prefix ),
+				),
+			);
+			if ( strpos( $hook_suffix, '/modules/aioseop_sitemap' ) !== false ) {
+				wp_register_script( 'aioseop-chosen', AIOSEOP_PLUGIN_URL . 'js/lib/chosen.jquery.min.js', array( 'jquery', 'jquery-ui-autocomplete' ), AIOSEOP_VERSION );
+				wp_register_style( 'aioseop-chosen', AIOSEOP_PLUGIN_URL . 'css/lib/chosen.min.css', array(), AIOSEOP_VERSION );
+				wp_enqueue_script( 'aioseop-chosen' );
+				wp_enqueue_style( 'aioseop-chosen' );
+				$deps[] = 'aioseop-chosen';
+				$options['l10n'] = array(
```

Comment:
Wouldn't it be more optimal & secure for clients/browsers to handle only the data that is needed, rather than sending/localizing the whole database options? If it is just translated text & a couple other values, then it would probably be better to just use those for localizing.

Reply:
I'm not sure what you mean. `$options` is a different variable here

Reply:
I was mistaken. I thought I saw an `$aioseop_options` variable, and the variable name threw me off.